prow/job/sippy.py

```diff
@@ -228,3 +235,72 @@ def last2weeks(self):
     def last4weeks(self):
         last4weeks = self._now - timedelta(days=28)
         return last4weeks.strftime(self.starttime_format)
+
+
+class DataAnalyzer():
+
+    def __init__(self, payload: dict):
+        if payload:
+            self._data_set = payload
+        else:
+            raise ValueError("payload is empty")
+
+    def is_component_readiness_status_green(self):
+        '''
+          Check if any component has attribute `regressed_tests` and status >= -1
+        '''
+        logger.info('-' * 50)
+        logger.info("Start to analyze component readiness status")
+
+        issued_comps = []
+        rows = self._data_set.get("rows")
+        if len(rows):
+            for comp in rows:
+                name = comp.get("component")
+                cols = comp.get("columns")
+                for col in cols:
+                    status = col.get("status")
+                    if status <= -2 or "regressed_tests" in col:
+                        regressed_tests = col.get("regressed_tests")
+                        logger.warning(
+                            f"{name} has regressed tests:\n{regressed_tests}\n")
+                        issued_comps.append(name)
+
+        if not issued_comps:
+            logger.info("No issued component found")
+
+        logger.info("Component readiness status is analyzed")
+
+        return len(issued_comps) == 0
+
+    def is_variants_status_green(self, expected_variants: [] = None):
+        '''
+          Check if the current pass percentage > or ~= previous pass percentage of the variants
+        '''
+
+        logger.info('-' * 50)
+        logger.info(
+            f"Start to analyze variants status {expected_variants if expected_variants else ''}")
+
+        issued_variants = []
+        if len(self._data_set):
+            for variant in self._data_set:
+                name = variant.get("name")
+                if expected_variants and name not in expected_variants:
+                    continue
+                current_pass_percentage = variant.get(
+                    "current_pass_percentage")
+                previous_pass_percentage = variant.get(
+                    "previous_pass_percentage")
+                net_improvement = variant.get("net_improvement")
+                if net_improvement < -1:
```

Comment:
@rioliu-rh `net_improvement < -1` to `net_improvement < 0` is better.
if net_improvement is -0.5, it should be degraded.

Reply:
where is the definition of the degraded condition, the diff b/w current and previous is less than 1 looks ok. it won't be displayed with degraded. e.g.
```
{
        "id": 0,
        "name": "assisted",
        "current_pass_percentage": 88.23529411764706,
        "current_runs": 17,
        "current_passes": 15,
        "current_fails": 2,
        "previous_pass_percentage": 88.88888888888889,
        "previous_runs": 18,
        "previous_passes": 16,
        "previous_fails": 2,
        "net_improvement": -0.6535947712418301
    }
```

Reply:
there are 3 kinds of variant status on dashboard. 
- increased
- decreased
- micro diff

for 3rd one, I suppose the condition is 0 < abs(net_improvement) < 1, but I found another variant does not follow this rule
```
{
        "id": 0,
        "name": "hypershift",
        "current_pass_percentage": 85.88709677419355,
        "current_runs": 248,
        "current_passes": 213,
        "current_fails": 35,
        "previous_pass_percentage": 87.5,
        "previous_runs": 256,
        "previous_passes": 224,
        "previous_fails": 32,
        "net_improvement": -1.6129032258064515
    },
```
this one is considered as micro diff on dashboard. 
<img width="192" alt="image" src="https://github.com/openshift/release-tests/assets/85483837/a5f51ca7-9505-4373-8cf1-48a3893cb7a1">

Reply:
xref: https://github.com/openshift/sippy/blob/72a5f3e16bc99483db09155707ad14280c3a7554/sippy-ng/src/components/PassRateIcon.js#L16

Reply:
updated the code with following logic
```
if net_improvement < 0 and abs(net_improvement) >= threshold:
```